go/config/config.go

```diff
@@ -34,18 +34,86 @@ func (addr *NodeAddress) toProtobuf() *protobuf.NodeAddress {
 	return &protobuf.NodeAddress{Host: addr.Host, Port: uint32(addr.Port)}
 }
 
+// ServiceType represents the types of AWS services that can be used for IAM authentication.
+type ServiceType int
+
+const (
+	// Elasticache represents Amazon ElastiCache service.
+	Elasticache ServiceType = iota
+	// MemoryDB represents Amazon MemoryDB service.
+	MemoryDB
+)
```

Comment:
Should we follow the existing CamelCase convention?

```suggestion
const (
	// Amazon ElastiCache service.
	ElastiCache ServiceType = iota
	// Amazon MemoryDB service.
	MemoryDb
)
```